codebase/superdarn/src.bin/tk/tool/fit_median_filter.1.0/fit_median_filter.c

```diff
@@ -0,0 +1,322 @@
+        index_list[8]=get_index(bm,ch,rng+1,t[bm][ch]+1,maxbm,maxch,maxrng);
+        
+        // corners
+        if (t[bm][ch]==0 && rng==0)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[2]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[8]];
+        else if (t[bm][ch]==0 && rng==maxrng-1)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[7]];
+        else if (t[bm][ch]==nrec[bm][ch] && rng==0)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[2]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[5]];
+        else if (t[bm][ch]==nrec[bm][ch] && rng==maxrng-1)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[4]];
+        
+        // edges
+        else if (t[bm][ch]==0)
+          sum = 2*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[2]] + qflgs->value[index_list[6]] 
+              + qflgs->value[index_list[7]]   + qflgs->value[index_list[8]];
+        else if (t[bm][ch]==nrec[bm][ch])
+          sum = 2*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[2]] + qflgs->value[index_list[3]] 
+              + qflgs->value[index_list[4]]   + qflgs->value[index_list[5]];
+        else if (rng==0)
+          sum = 2*qflgs->value[index_list[0]] + qflgs->value[index_list[2]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[5]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[8]];
+        else if (rng==maxrng-1)
+          sum = 2*qflgs->value[index_list[0]] + qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[4]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[7]];
+        
+        // all other cells
+        else sum = qflgs->value[index_list[0]] + qflgs->value[index_list[1]] 
+                 + qflgs->value[index_list[2]] + qflgs->value[index_list[3]] 
+                 + qflgs->value[index_list[4]] + qflgs->value[index_list[5]] 
+                 + qflgs->value[index_list[6]] + qflgs->value[index_list[7]] 
+                 + qflgs->value[index_list[8]];
+        
+        // Remove record if median=0 (sum of qflgs < 5)
+        if (sum < 5) fit->rng[rng].qflg=0;
+      }
+    }
+    t[bm][ch]++;
+    
+    
+    // Set origin fields
+    prm->origin.code = 1; // file was not generated at radar site
+    ctime = time((time_t) 0);
+    if (RadarParmSetOriginCommand(prm,command) == -1) 
+    {
+        fprintf(stderr,"Error: cannot set Origin Command\n");
+        free_parameters(prm, fit, fp, qflgs);
+        exit(-1);
+    }
+
+    strcpy(tmstr,asctime(gmtime(&ctime)));
```

Comment:
check return value or errno

Reply:
Again, could you please make a suggestion on how I should change this? I just copied this line from elsewhere in RST. Thanks :smile: